db/column_family.h

```diff
@@ -427,8 +428,19 @@ class ColumnFamilyData {
     return internal_comparator_;
   }
 
-  const IntTblPropCollectorFactories* int_tbl_prop_collector_factories() const {
-    return &int_tbl_prop_collector_factories_;
+  IntTblPropCollectorFactoryRange int_tbl_prop_collector_factories(
+      bool enable_blob_property_collector) const {
+    assert(!int_tbl_prop_collector_factories_.empty());
+    assert(dynamic_cast<const BlobTablePropertiesCollectorFactory*>(
+        int_tbl_prop_collector_factories_.front().get()));
```

Comment:
I feel the restriction that BlobTablePropCollectorFactory has to be the first one is not so obvious.

Reply:
It's added before the user-specified ones (see lines 558-559 in `column_family.cc`) so it's guaranteed to be at the head of the list. I can add a comment though to point this out.